modules/services/xcape.nix

```
@@ -63,6 +63,8 @@ in {
         Description = "xcape";
         After = [ "graphical-session-pre.target" ];
         PartOf = [ "graphical-session.target" ];
+      } // mkIf (config.home.keyboard != null) {
+        After = [ "graphical-session-pre.target" "setxkbmap.service" ];
       };
```

```suggestion
        Description = "xcape";
        After = [ "graphical-session-pre.target" ];
        PartOf = [ "graphical-session.target" ];
        After = mkIf (config.home.keyboard != null) [ "graphical-session-pre.target" "setxkbmap.service" ];
      }
```

Using `mkMerge` or `mkIf` with the merge operator (`//`) does not work.

Either set `Unit = mkMerge [ { .... } (mkIf ... { }) ]`; or set `After = mkIf ... value;`.

---

You are right. It did "work" by coincidence, because `Description` and `PartOf` turn out non-critical.

Are you sure your suggested change works? I get an error 

    error: attribute 'After' at /home/akho/home-manager/modules/services/xcape.nix:66:9 already defined at /home/akho/home-manager/modules/services/xcape.nix:64:9

`mkMerge` works, I think. Amended.

My first module, thank you for correcting me.